test/core/test/jest-expect.test.ts

```diff
@@ -375,6 +368,18 @@ describe('jest-expect', () => {
       },
     ])
   })
+
+  it('toThrow didn\'t throw', () => {
+    expect(() => {
+      expect(async () => {
+      }).toThrow(Error)
```

Comment:
Async error should be with `.rejects`:

```ts
await expect(async () => {}).rejects.toThrow(Error)
```

Reply:
The results are consistent with `jest`
<img width="545" alt="image" src="https://github.com/vitest-dev/vitest/assets/29533304/6bff309e-f10b-4d42-a2d3-c171b553f65f">

Reply:
Yes, but it doesn't mean that code is correct. Function didn't finish running yet

Reply:
`toThrow` receives a Promise, it doesn't wait it to resolve to _ACTUALLY_ check that error didn't happen

If you throw an error there, you will get unhandled promise error, not `toThrow` error

Reply:
This test case actually comes from [here](https://github.com/vitest-dev/vitest/issues/3941), which is exactly what I wanted to test

Reply:
> Async error should be with `.rejects`:
> 
> ```ts
> await expect(async () => {}).rejects.toThrow(Error)
> ```

I got an error
<img width="1023" alt="image" src="https://github.com/vitest-dev/vitest/assets/29533304/fbb06dbf-7bca-4e8c-82cf-b3bbf55383ac">

Reply:
> This test case actually comes from https://github.com/vitest-dev/vitest/issues/3941, which is exactly what I wanted to test

Well, then your error description is not enough "but it didn't threw" doesn't really tell much. In the current example there are two errors: 1) async wasn't awaited, 2) error wasn't thrown

> I got an error

Please, read how to test asynchronous code: https://vitest.dev/api/expect.html#resolves